pkg/systemd/services/service-details.jsx

```diff
@@ -291,15 +291,21 @@
 
     onOnOffSwitch() {
         if (this.props.unit.UnitFileState === "enabled") {
-            this.unitFileAction("DisableUnitFiles", undefined);
+            let promise;
             if (this.props.unit.ActiveState === "active" || this.props.unit.ActiveState === "activating")
-                this.unitAction("StopUnit");
-            if (this.props.unit.ActiveState === "failed")
-                this.unitAction("ResetFailedUnit", []);
+                promise = this.unitAction("StopUnit");
+            else if (this.props.unit.ActiveState === "failed")
+                promise = this.unitAction("ResetFailedUnit", []);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17999-20221206-150957-1a6e9961-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)